### PR TITLE
Update diff test for e2e and interactive mode

### PR DESCRIPTION
**Pull request by OliviaLynn**
## Change Description
For #22 

Makes diff test e2e and adds interactive mode, which provides more control, including the option to force regenerate goldens.

## Code Quality
- [x] I have read the Contribution Guide
- [x] My code follows the code style of this project
- [x] My code builds (or compiles) cleanly without any errors or warnings
- [x] My code contains relevant comments and necessary documentation

### CODE REVIEW

File: tests/superphot_plus/diff_test.py [hunk 1]
```diff
@@ -1,109 +1,261 @@
...
+        raise ValueError("Attempted to delete directory that is not designated with 'temp'.")
+    for file in os.listdir(temp_dir):
+        os.remove(Path(temp_dir, file))
+    os.rmdir(temp_dir)
 
 
 def test_diffs(tmp_path):
-    """Make golden files if they do not yet exist; otherwise, make new
-    results and compare with goldens.
+    """Check new results against pre-generated goldens file.
```
Comment (jeremykubica) [inline tests/superphot_plus/diff_test.py:207]: This isn't called anywhere? Should this be removed?

**Review by jeremykubica — CHANGES_REQUESTED** [review]
Only one requested change (removing the check that the files have the same number of lines). The rest are just questions and suggestions.

File: tests/superphot_plus/diff_test.py (hunk 1 above)
Comment (OliviaLynn) [reply tests/superphot_plus/diff_test.py:207]: This is the one that will be run by pytest (by virtue of it being in the tests/ directory and the function beginning with "test")

Comment (jeremykubica) [reply tests/superphot_plus/diff_test.py:207]: Makes sense. Thanks

File: tests/superphot_plus/diff_test.py [hunk 2]
```diff
@@ -1,109 +1,261 @@
...
+    assert check_goldens_exist(goldens_dir)
+    generate_files(lightcurve, tmp_path)
+    assert compare_directories(goldens_dir, tmp_path)
+
+
+def interactive_test_diffs():
+    """Tests diffs interactively, including a prompt to force regeneration of goldens."""
+    data_dir = Path("tests", "data")
+    goldens_dir = Path(data_dir, "goldens")
+    temp_dir = Path(data_dir, "temp_diff")
```
[the remark this answers is not in the document]
Comment (OliviaLynn) [reply tests/superphot_plus/diff_test.py:227]: Updated

[the remark this answers is not in the document]
Comment (OliviaLynn) [reply tests/superphot_plus/diff_test.py:227]: After our conversation about the benefits of using standard functions vs would-be-nice to keep the option of keeping temp files around a little longer if investigating a failed test, I started working on something with [tempfile.mkdtemp()](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp) (which places the responsibility of clean up on the user). 

Realized I was falling down a rabbit hole, so opened [this issue](https://github.com/lincc-frameworks/superphot-plus/issues/127) to get this PR in and potentially carry on with temp dirs later.

File: tests/superphot_plus/diff_test.py [hunk 3]
```diff
@@ -1,109 +1,262 @@
...
+    ----------
+    goldens_dir : pathlib Path
+        The directory containing our goldens.
+    temp_results_dir : pathlib Path
+        The directory containing the temporary results files we've just generated.
+
+    Returns
+    -------
+    boolean
+        True if files in directories are suficiently similar; False otherwise.
```
Comment (jeremykubica) [inline tests/superphot_plus/diff_test.py:109]: suficiently -> sufficiently

File: tests/superphot_plus/diff_test.py [hunk 4]
```diff
@@ -1,109 +1,262 @@
...
+        The file to compare (a file of this name should exist in each dir).
+    goldens_dir : pathlib Path
+        The directory containing our goldens.
+    temp_results_dir : pathlib Path
+        The directory containing the temporary results files we've just generated.
+
+    Returns
+    -------
+    boolean
+        True if files are suficiently similar; False otherwise."""
```
Comment (jeremykubica) [inline tests/superphot_plus/diff_test.py:149]: suficiently -> sufficiently

**Review by jeremykubica — APPROVED** [review]

**Pull request merged**